### PR TITLE
Add OSG 23 development images (SOFTWARE-5673)

### PR DESCRIPTION
**Pull request by brianhlin**
Tested that the expected jobs are created here https://github.com/brianhlin/docker-software-base/actions/runs/6240846201

### CODE REVIEW

File: .github/workflows/build-container.yml [hunk 1]
```diff
@@ -36,7 +36,18 @@ jobs:
           - image: 'nvidia/cuda:11.8.0-runtime-rockylinux8'
             tag_str: 'cuda_11_8_0'
         repo: ['development', 'testing', 'release']
-        series: ['3.6']
+        series: ['3.6', '23']
+        # Exclude EL7 for OSG 23 builds
+        # FIXME: exclude testing and release for OSG 23 builds
```
Comment (matyasselmeci) [inline .github/workflows/build-container.yml:41]: uh... can you do this?

File: Dockerfile [hunk 2]
```diff
@@ -23,7 +23,12 @@ RUN \
        YUM_PKG_NAME="yum-utils"; \
     fi && \
     yum distro-sync -y && \
-    yum -y install http://repo.opensciencegrid.org/osg/${OSG_RELEASE}/osg-${OSG_RELEASE}-el${DVER}-release-latest.rpm \
+    if [[ $OSG_RELEASE =~ ^[0-9][0-9]$ ]]; then \
+       OSG_URL=http://repo.opensciencegrid.org/osg/${OSG_RELEASE}-main/osg-${OSG_RELEASE}-main-el${DVER}-release-latest.rpm; \
+    else \
+       OSG_URL=http://repo.opensciencegrid.org/osg/${OSG_RELEASE}/osg-${OSG_RELEASE}-el${DVER}-release-latest.rpm; \
+    fi && \
```
Comment (matyasselmeci) [inline Dockerfile:30]:
```suggestion
    if [[ $OSG_RELEASE =~ ^[0-9][0-9]$ ]]; then \
       OSG_URL=https://repo.opensciencegrid.org/osg/${OSG_RELEASE}-main/osg-${OSG_RELEASE}-main-el${DVER}-release-latest.rpm; \
    else \
       OSG_URL=https://repo.opensciencegrid.org/osg/${OSG_RELEASE}/osg-${OSG_RELEASE}-el${DVER}-release-latest.rpm; \
    fi && \
```

File: .github/workflows/build-container.yml (hunk 1 above)
Comment (brianhlin) [reply .github/workflows/build-container.yml:41]: Yup! https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#excluding-matrix-configurations You can also do includes

**Comment by brianhlin** [comment]
I initially marked this as `do-not-merge` because of the condor RPM signing issue but I don't think it affects the base

**Comment by brianhlin** [comment]
This appears to work https://github.com/brianhlin/docker-software-base/actions/runs/6251442139/job/16972582566

**Pull request merged**